### ISSUE TITLE
How to generate dataset for training text recognition?

### ISSUE THREAD

**Issue by gachiemchiep** [PaddlePaddle/PaddleOCR#2804]
Hello. 
As the title said. 
How did you generate a dataset for training text recognition. 
I want to train text recognition for Japanese.

Thanks.

**Comment by littletomatodonkey** [comment]
You need to prepare for the corpus, char_list file and backgraound images, and then generate the images using some tools such as TextRenderer.

**Comment by gachiemchiep** [comment]
Hello @littletomatodonkey. 
Thank you for your answer. 
The corpus, char_list files are easy to prepare. 
 I'm having difficulty preparing background images. 
Which dataset are you using for background images?

**Comment by littletomatodonkey** [comment]
ImageNet is a good choice, of course, you can also try coco dataset.

**Comment by gachiemchiep** [comment]
@littletomatodonkey 
Thank you for your advice. I will try.

**Comment by gachiemchiep** [comment]
@littletomatodonkey 
I tried to use StyleText to generate some Japanese text. But a lot of images are wrong. 
Do you know how  to add new language to StyleText ? 

Here is some images that ware generated by StyleText. 

![5](https://user-images.githubusercontent.com/2300329/118766672-b5968a80-b8b7-11eb-9e15-cef24dcadc0d.png)
![1](https://user-images.githubusercontent.com/2300329/118766292-34d78e80-b8b7-11eb-924c-28bc4c768b3a.png)
![7](https://user-images.githubusercontent.com/2300329/118766428-5d5f8880-b8b7-11eb-90ac-68e68ec8f994.png)
![6](https://user-images.githubusercontent.com/2300329/118766442-63556980-b8b7-11eb-8602-93d6461ad6e1.png)

**Comment by weisy11** [comment]
StyleText only support Chinese, Korean and English. We will release training code once it's ready.

**Comment by gachiemchiep** [comment]
@weisy11  
Thank you. I will wait for your code.

**Comment by gachiemchiep** [comment]
Hello @weisy11 @littletomatodonkey 

Is the training dataset for Japanese are available for download?

I tried [TextRecognitionDataGenerator](https://github.com/Belval/TextRecognitionDataGenerator) to generate 13GB of sample image. 

![96402](https://user-images.githubusercontent.com/2300329/125720360-74d97c06-54d3-4561-b31d-b2966197c918.jpg)


Unfortunately the trained model didn't work well on real images. 
Here's a snip of training log

```bash
[2021/07/14 18:50:58] root INFO: epoch: [6/500], iter: 31980, lr: 0.001000, loss: 0.383117, acc: 0.941406, norm_edit_dis: 0.980974, reader_cost: 0.00349 s, batch_cost: 1.03857 s, samples: 2560, ips: 246.49245
[2021/07/14 18:51:30] root INFO: epoch: [6/500], iter: 31990, lr: 0.001000, loss: 0.368604, acc: 0.939453, norm_edit_dis: 0.980236, reader_cost: 0.00349 s, batch_cost: 1.03370 s, samples: 2560, ips: 247.65304
[2021/07/14 18:52:02] root INFO: epoch: [6/500], iter: 32000, lr: 0.001000, loss: 0.332468, acc: 0.941406, norm_edit_dis: 0.980410, reader_cost: 0.00358 s, batch_cost: 1.03814 s, samples: 2560, ips: 246.59434
[2021/07/14 18:52:11] root INFO: cur metric, acc: 0.039525691699604744, norm_edit_dis: 0.2213243817739866, fps: 182.44605328506736
[2021/07/14 18:52:11] root INFO: best metric, acc: 0.08036890645586298, norm_edit_dis: 0.24336490352242524, fps: 180.10117454714774, best_epoch: 4
[2021/07/14 18:52:44] root INFO: epoch: [6/500], iter: 32010, lr: 0.001000, loss: 0.323103, acc: 0.941406, norm_edit_dis: 0.982170, reader_cost: 0.00455 s, batch_cost: 1.09996 s, samples: 2560, ips: 232.73591
[2021/07/14 18:53:16] root INFO: epoch: [6/500], iter: 32020, lr: 0.001000, loss: 0.358400, acc: 0.937500, norm_edit_dis: 0.979244, reader_cost: 0.00532 s, batch_cost: 1.07220 s, samples: 2560, ips: 238.76070
[2021/07/14 18:53:49] root INFO: epoch: [6/500], iter: 32030, lr: 0.001000, loss: 0.392107, acc: 0.931641, norm_edit_dis: 0.978685, reader_cost: 0.00378 s, batch_cost: 1.04085 s, samples: 2560, ips: 245.95294
[2021/07/14 18:54:21] root INFO: epoch: [6/500], iter: 32040, lr: 0.001000, loss: 0.403803, acc: 0.929688, norm_edit_dis: 0.977142, reader_cost: 0.00381 s, batch_cost: 1.07060 s, samples: 2560, ips: 239.11798

```

**Comment by anguoyang** [comment]
@gachiemchiep how thing going of your training? have you recognized the japanese text better?

**Comment by gachiemchiep** [comment]
@anguoyang
 I end up using the realdata to train my model

**Comment by anguoyang** [comment]
thanks for your reply, just wonder how about the pretrained Japanese model, is it good enough for production? @gachiemchiep

**Comment by gachiemchiep** [comment]
@anguoyang 
Unfortunately, it wasn't good enough.

**Comment by thanh15dt2** [comment]
> @anguoyang Unfortunately, it wasn't good enough.

I am using a lot of synthesis data for impoving accuracy. My purpose is just ocr in somewhere like banner which is clear text, i hope this is good enough for production.  Can you give me some advice about it? thanks

**Comment by Sayaka91** [comment]
Thanks for your answer.
I'm working with data extraction from japanese receipts.
I tried using paddleocr japanese model but the result is not good.
So i'll try training model with a custom dataset.
Cảm ơn bạn


Vào Th 3, 23 thg 5, 2023 vào lúc 15:11 vugia truong <
***@***.***> đã viết:

**Comment by Sayaka91** [comment]
Sorry but when I used this tool (PPOCCRLabel), the recognized text is
always en English.
I can't change *lang *to Japanese.
Do you know why this happens?

Vào Th 3, 23 thg 5, 2023 vào lúc 18:08 vugia truong <
***@***.***> đã viết:

**Comment by gachiemchiep** [comment]
@Sayaka91
You have to change the model. It is written on the manual 

Model language switching: Changing the built-in model language is supportable by clicking "PaddleOCR"-"Choose OCR Model" in the menu bar. Currently supported languages​include French, German, Korean, and Japanese. For specific model download links, please refer to [PaddleOCR Model List](https://github.com/PaddlePaddle/PaddleOCR/blob/develop/doc/doc_en/models_list_en.md#multilingual-recognition-modelupdating)

**Comment by Sayaka91** [comment]
@gachiemchiep 
I changed selected model to Japanese but got this error.
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/49243283/bf3efeca-e325-4a1e-8cd5-a377cd92cfc7)

**Comment by gachiemchiep** [comment]
@Sayaka91 
Yes you can use your customized models. 

Add value of det_model_dir, rec_model_dir, cls_model_dir to this line
https://github.com/PaddlePaddle/PaddleOCR/blob/013870d9bc963f7508d5a76f81da872b84471a71/PPOCRLabel/PPOCRLabel.py#L97

**Comment by Sayaka91** [comment]
@gachiemchiep 
Thanks for your help. 
But you finetuned the japanese ocr model based on paddleocr, right? 
Can you send me your dictionary? I got the accuracy 0 instead of the pretrained model accuray

**Comment by gachiemchiep** [comment]
@Sayaka91 

For synthetic data, you will need few millions image.  
You could use https://github.com/Belval/TextRecognitionDataGenerator or  https://github.com/clovaai/synthtiger for generating synthetic images. 

Even-though the recognition model is labeled as "lite", it actually works very well. 
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/configs/rec/multi_language/rec_japan_lite_train.yml
So if you don't have any new-character (for example: ㈱), it doesn't need to re-train the entire model on Synthetic images. 

You can check new-character by comparing with default character dictionary for japanese model here:
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppocr/utils/dict/japan_dict.txt

For real data, we use like 1 hundred sample per character.

**Comment by Sayaka91** [comment]
@gachiemchiep 
Thank you very much for your help. 
I changed the dictionary with some new words. The recognition result is really bad although I trained the model with ~40k synthetic data. I'll give a try with your recommend.
